### PR TITLE
fix complex type slice decoding

### PR DESCRIPTION
**Pull request by jsign**
This PR fixes some mistakes in the way complex types are decoded, mostly around wrong constants being used.

I've also noticed that this complex type decoding isn't working correctly for slices that actually contain elements. (i.e: now things work for empty slices). I can explain better in Matrix or in a further issue to keep track of it.

### CODE REVIEW

File: src/deserialize.zig [hunk 1]
```diff
@@ -120,19 +120,22 @@ pub fn deserialize(comptime T: type, serialized: []const u8, out: *T) !usize {
...
-                if (serialized[0] < rlpByteListShortHeader) {
-                    return try deserialize(ptr.child, serialized[0..1], &out.*[0]);
-                } else if (serialized[0] < rlpByteListLongHeader) {
-                    const size = @as(usize, serialized[0] - rlpByteListShortHeader);
+                if (serialized[0] < rlpListShortHeader) {
+                    return error.NotAnRLPList;
+                }
+
+                if (serialized[0] < rlpListLongHeader) {
+                    const size = @as(usize, serialized[0] - rlpListShortHeader);
```
Comment (jsign) [inline src/deserialize.zig:128]: I'm not entirely sure why the current (previous) code was that way -- maybe a copy-paste from other case?

File: src/deserialize.zig [hunk 2]
```diff
@@ -364,3 +367,18 @@ test "detects an invalid length serialization" {
...
+            storage_keys: [][32]u8,
+        },
+    };
+
+    var buf: [128]u8 = undefined;
+    const rlp = try std.fmt.hexToBytes(&buf, "c1c0");
+
+    var out: StrippedTxn = undefined;
+    _ = try deserialize(StrippedTxn, rlp, &out);
+}
```
Comment (jsign) [inline src/deserialize.zig:384]: Test that failed (before) and passes (after).

It's a narrowed-down version on how I found this bug. Basically, an empty access list.

File: src/deserialize.zig (hunk 1 above)
Comment (gballet) [reply src/deserialize.zig:128]: or I thought of a different type of list, but nothing comes to mind right now so it's probably an error indeed.

**Review by gballet — APPROVED** [review]
LGTM thanks for catching that.

**Pull request merged**